govcd/vm.go

```diff
@@ -241,50 +241,52 @@ func (vm *VM) ChangeNetworkConfig(networks []map[string]interface{}, ip string)
 		return Task{}, fmt.Errorf("error refreshing VM before running customization: %v", err)
 	}
 
-	networksection, err := vm.GetNetworkConnectionSection()
-
-	for index, network := range networks {
-		// Determine what type of address is requested for the vApp
-		ipAllocationMode := "NONE"
-		ipAddress := "Any"
-
-		// TODO: Review current behaviour of using DHCP when left blank
-		if ip == "dhcp" || network["ip"].(string) == "dhcp" {
-			ipAllocationMode = "DHCP"
-		} else if ip == "allocated" || network["ip"].(string) == "allocated" {
-			ipAllocationMode = "POOL"
-		} else if ip == "none" || network["ip"].(string) == "none" {
-			ipAllocationMode = "NONE"
-		} else if ip != "" {
-			ipAllocationMode = "MANUAL"
-			// TODO: Check a valid IP has been given
-			ipAddress = ip
-		} else if network["ip"].(string) != "" {
-			ipAllocationMode = "MANUAL"
-			// TODO: Check a valid IP has been given
-			ipAddress = network["ip"].(string)
-		} else if ip == "" {
-			ipAllocationMode = "DHCP"
+	networkSection, err := vm.GetNetworkConnectionSection()
+
+	for _, network := range networks {
+		for index, networkConnection := range networkSection.NetworkConnection {
```

Comment:
This double loop is tricky to read. The fact that `networks` is a map doesn't help either. Please add a human-oriented comment above that explains what is happening here.

Reply:
Added requested comment